src/core/layout/qgslayoutexporter.cpp

```diff
@@ -1275,6 +1281,50 @@ void QgsLayoutExporter::preparePrintAsPdf( QgsLayout *layout, QPagedPaintDevice
 
   updatePrinterPageSize( layout, device, firstPageToBeExported( layout ) );
 
+  // force a non empty title to avoid invalid (according to specification) PDF/X-4
+  const QString title = layout->project()->metadata().title().isEmpty() ?
+                        fi.baseName() : layout->project()->metadata().title();
+
+  QPdfWriter *pdfWriter = static_cast<QPdfWriter *>( device );
```

Comment:
```suggestion
  QPdfWriter *pdfWriter = qobject_cast<QPdfWriter *>( device );
```

Reply:
I changed method signature because device is always a QPdfWriter c2297fcc5b2

There is still a static_cast on a special workaround that is now fixed.